core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java

```diff
@@ -175,9 +179,29 @@ private Map<PlanNodeId, SplitSource> visitScanAndFilter(TableScanNode node, Opti
             }
 
             Constraint constraint = filterPredicate
-                    .map(predicate -> filterConjuncts(plannerContext.getMetadata(), predicate, expression -> !DynamicFilters.isDynamicFilter(expression)))
-                    .map(predicate -> new LayoutConstraintEvaluator(plannerContext, typeAnalyzer, session, typeProvider, node.getAssignments(), predicate))
-                    .map(evaluator -> new Constraint(TupleDomain.all(), evaluator::isCandidate, evaluator.getArguments())) // we are interested only in functional predicate here, so we set the summary to ALL.
+                    .map(predicate -> {
+                        Metadata metadata = plannerContext.getMetadata();
+                        predicate = filterConjuncts(metadata, predicate, expression -> !isDynamicFilter(expression) && isDeterministic(expression, metadata));
+                        DomainTranslator.ExtractionResult decomposedPredicate = DomainTranslator.getExtractionResult(plannerContext, session, predicate, typeProvider);
+                        ConnectorExpressionTranslator.ConnectorExpressionTranslation expressionTranslation = ConnectorExpressionTranslator.translateConjuncts(
+                                session,
+                                decomposedPredicate.getRemainingExpression(),
+                                typeProvider,
+                                plannerContext,
+                                typeAnalyzer);
+                        Map<String, ColumnHandle> connectorExpressionAssignments = node.getAssignments()
+                                .entrySet().stream()
+                                .collect(toImmutableMap(entry -> entry.getKey().getName(), Map.Entry::getValue));
+                        // TODO Constraint.predicate is supposed not to cover Constraint.expression, but not Constraint.expression
```

Comment:
Typo? Confusing / conflicting references to `Constraint.expresssion` in the comment

Reply:
thanks, fixed